### PR TITLE
Umcs 147/feature/applepay example

### PR DESCRIPTION
**Pull request by Ryouzanpaku**
Add applepay example.

### CODE REVIEW

File: examples/Applepay/Constants.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * This file defines the constants needed for the card example.
```
Comment (sixer1182) [inline examples/Applepay/Constants.php:3]:
```suggestion
 * This file defines the constants needed for the Apple Pay example.
```

File: examples/Applepay/Controller.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * This is the controller for the Card example.
```
Comment (sixer1182) [inline examples/Applepay/Controller.php:3]:
```suggestion
 * This is the controller for the Apple Pay example.
```

File: examples/Applepay/Controller.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @link  https://docs.unzer.com/
+ *
+ * @author  Simon Gabriel <development@unzer.com>
```
Comment (sixer1182) [inline examples/Applepay/Controller.php:22]: @Ryouzanpaku

File: examples/Applepay/index.php [hunk 4]
```diff
@@ -0,0 +1,240 @@
+<?php
+/**
+ * This file provides an example implementation of the Card payment type.
```
Comment (sixer1182) [inline examples/Applepay/index.php:3]:
```suggestion
 * This file provides an example implementation of the Apple Pay payment type.
```

File: examples/Applepay/index.php [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @link  https://docs.unzer.com/
+ *
+ * @author  Simon Gabriel <development@unzer.com>
```
Comment (sixer1182) [inline examples/Applepay/index.php:21]: It wasn't me

File: examples/Applepay/Constants.php [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @link  https://docs.unzer.com/
+ *
+ * @author  Simon Gabriel <development@unzer.com>
```
Comment (sixer1182) [inline examples/Applepay/Constants.php:21]: It wasn't me

File: examples/Applepay/index.php [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+    if (!window.ApplePaySession || !ApplePaySession.canMakePayments() || !ApplePaySession.supportsVersion(APPLE_PAY_VERSION)) {
+        handleError('This device does not support Apple Pay version 6!', APPLE_PAY_VERSION);
+    } else {
+        var unzerApplePayInstance = unzerInstance.ApplePay();
+
+        var applePayInformationObject = unzerApplePayInstance.createApplePaySessionRequest({
+            countryCode: 'DE',
+            currencyCode: 'EUR',
+            supportedNetworks: ['visa', 'masterCard'],
+            merchantCapabilities: ['supports3DS', 'supportsCredit', 'supportsDebit'],
```
Comment (sixer1182) [inline examples/Applepay/index.php:133]: Do we support all of those?

File: examples/Applepay/merchantvalidation.php [hunk 8]
```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * This is the controller for the Card example.
```
Comment (sixer1182) [inline examples/Applepay/merchantvalidation.php:3]: see above

File: examples/Applepay/merchantvalidation.php [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @link  https://docs.unzer.com/
+ *
+ * @author  Simon Gabriel <development@unzer.com>
```
Comment (sixer1182) [inline examples/Applepay/merchantvalidation.php:22]: see above

**Review by sixer1182 — CHANGES_REQUESTED** [review]
Apart from some minor issues: LGTM
Let one of the other guys check as well, please.

File: examples/Applepay/index.php [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+
+            onShippingMethodSelectedCallback: (session, event) => {
+                var status = ApplePaySession.STATUS_SUCCESS;
+                var newTotal = {
+                    'label': 'Total amount',
+                    'amount': '12.99',
+                    'type': 'final'
+                }
+                var newLineItems =[]; // What does this mean?
+                session.completeShippingMethodSelection(status, newTotal, newLineItems);
```
Comment (mghazzouz) [inline examples/Applepay/index.php:191]: Maybe remove the comment, or just do the call without the newLineItems parameter
Here you can find, what it is for: https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession/1778024-completeshippingmethodselection

File: examples/Applepay/index.php [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+            merchantCapabilities: ['supports3DS', 'supportsCredit', 'supportsDebit'],
+            totalLabel: '"Unzer UI Demo (Card is not charged)"',
+            totalAmount: '0.0',
+
+            onMerchantValidationCallback: (session, event) => {
+                $.post('./merchantvalidation.php', JSON.stringify({"merchantValidationUrl": event.validationURL}), null, 'json')
+                    .done(function (validationResponse) {
+                        // Contains Applepay session on succsess.
+                        session.completeMerchantValidation(validationResponse);
+                    })
```
Comment (mghazzouz) [inline examples/Applepay/index.php:142]: I think "merchant session" ist the right word here (not Applepay session)
see here: https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession/1778021-onvalidatemerchant

**Review by mghazzouz — CHANGES_REQUESTED** [review]
Looks also fine for me.
Have only small remarks, see my comments

File: examples/Applepay/index.php (hunk 10 above)
Comment (sixer1182) [reply examples/Applepay/index.php:191]: I don't think we need the `newLineItems` in the examples at all

**Review by mghazzouz — CHANGES_REQUESTED** [review]
Hi David, please also replace all 'var' with 'const' (if possible) or 'let'.

**Review by sixer1182 — DISMISSED** [review]

File: examples/Applepay/index.php (hunk 11 above)
Comment (Ryouzanpaku) [reply examples/Applepay/index.php:142]: changed it to "applepay merchant session" to make clear its from appleay.

File: examples/Applepay/index.php (hunk 10 above)
Comment (Ryouzanpaku) [reply examples/Applepay/index.php:191]: removed it for now.

**Review by sixer1182 — APPROVED** [review]

File: examples/Applepay/index.php (hunk 7 above)
Comment (sixer1182) [reply examples/Applepay/index.php:133]: Ah, this question is still open @Ryouzanpaku

**Review by sixer1182 — APPROVED** [review]
LGTM

File: examples/Applepay/index.php (hunk 7 above)
Comment (sixer1182) [reply examples/Applepay/index.php:133]: never mind

**Review by sixer1182 — APPROVED** [review]

**Review by mghazzouz — APPROVED** [review]

**Pull request merged**